lib/bots/versionbot.ts

```diff
@@ -390,6 +409,228 @@ export class VersionBot extends ProcBot {
     }
 
     /**
+     * Checks a freshly opened PR to see if there are any configured reviewers and maintainers.
+     * Should any valid reviewers and maintainers not already be assigned as reviewers on the PR, then
+     * a comment is posted on the PR directly asking those user logins to add themselves.
+     *
+     * @param _registration GithubRegistration object used to register the method
+     * @param event         ServiceEvent containing the event information ('pull_request' event)
+     * @returns             A void Promise once execution has finished.
+     */
+    protected addReviewers = (_registration: GithubRegistration, event: ServiceEvent): Promise<void> => {
+        const pr = event.cookedEvent.data.pull_request;
+        const head = event.cookedEvent.data.pull_request.head;
+        const owner = head.repo.owner.login;
+        const repo = head.repo.name;
+        let approvedMaintainers: string[];
+        let approvedReviewers: string[];
+
+        // Only when opening a new PR.
+        console.log(event.cookedEvent.data.action);
+        if (event.cookedEvent.data.action !== 'opened') {
+            return Promise.resolve();
+        }
+
+        this.logger.log(LogLevel.INFO, `Checking reviewers list for ${owner}/${repo}#${pr.number}`);
+
+        // Get the reviewers for the PR.
+        return this.retrieveConfiguration({
+            emitter: this.githubEmitter,
+            location: {
+                owner,
+                repo,
+                path: RepositoryFilePath
+            }
+        }).then((config: VersionBotConfiguration) => {
+            approvedMaintainers = this.stripPRAuthor((config || {}).maintainers || null, pr) || [];
+            approvedReviewers = this.stripPRAuthor((config || {}).reviewers || null, pr) || [];
+
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    number: pr.number,
+                    owner,
+                    repo
+                },
+                method: this.githubApi.pullRequests.get
+            });
+        }).then((pullRequest: GithubApiTypes.PullRequest) => {
+            // Look at the reviewers, create a list of configured reviewers that were not added
+            // when the PR was opened.
+            const assignedReviewers = _.map(pullRequest.requested_reviewers, (reviewer) => reviewer.login);
+            const configuredReviewers = _.uniq(_.unionWith(approvedReviewers, approvedMaintainers));
+            let missingReviewers = _.filter(configuredReviewers, (reviewer) => {
+                return !(_.find(assignedReviewers, (assignedReviewer) => (assignedReviewer === reviewer)) ||
```

Comment:
I'm pretty against adding logical operators on the next line, it leaves the previous one hanging with no context as to what's coming. To me, this is far clearer.